crates/l3/appchain-core-contract-client/src/interfaces/operator.rs

```diff
@@ -0,0 +1,103 @@
+use color_eyre::{eyre::eyre, Result};
+use common::LocalWalletSignerMiddleware;
+use common::{call_contract, invoke_contract};
+use starknet_core::types::{FieldElement, InvokeTransactionResult};
+use std::sync::Arc;
+
+pub struct Operator<'a> {
+    client: Arc<&'a LocalWalletSignerMiddleware>,
```

Comment:
same. It is not the client here. And can you get rid of the `Arc` or is it required for something?

Reply:
yeah so the structure followed was based on the existing implementation of ethereum contract clients, where Arc is passed for thread safety i believe, as the client passed from other repo(in this case madara/orchestrator) was with Arc::new(). should we keep this or can be optimised?

Reply:
But its content is an immutable reference. So it doesn't add any safety here. Does it?